ui/src/app/pages/homepage/data-access-requirements.tsx

```diff
@@ -281,22 +282,42 @@ const handleRasCallback = (code: string, spinnerProps: WithSpinnerOverlayProps)
     spinnerProps.showSpinner();
     await profileApi().linkRasAccount({ authCode: code, redirectUrl: buildRasRedirectUrl() });
     spinnerProps.hideSpinner();
+    reloadProfile();
   });
 
   return handler();
 };
 
-const selfBypass = async(spinnerProps: WithSpinnerOverlayProps, reload: Function) => {
+const selfBypass = async(spinnerProps: WithSpinnerOverlayProps, reloadProfile: Function) => {
   spinnerProps.showSpinner();
   await bypassAll(allModules, true);
   spinnerProps.hideSpinner();
-  reload();
+  reloadProfile();
 };
 
-const Refresh = () => <Button
+const syncExternalModules = async() => {
+  const aborter = new AbortController();
+  try {
+    await Promise.all([
+      profileApi().syncTwoFactorAuthStatus({signal: aborter.signal}),
```

Comment:
FYI: If multiple of these have changed in parallel, you'll likely get concurrent modification errors. Running them serially would be another solution, but would be ~3X as slow

Reply:
Good point.  I'll think about a solution in a follow-up PR.